src/utils/eventDispatcher.test.ts

```diff
@@ -0,0 +1,62 @@
+import EventDispatcher from './eventDispatcher';
+
+describe('EventDispatcher', () => {
+  test('should emit event', () => {
+    const eventDispatcher = new EventDispatcher();
+
+    let eventEmitted = false;
+    eventDispatcher.addEventListener('bar', () => {
+      eventEmitted = true;
+    });
+
+    eventDispatcher.emit('bar');
+    expect(eventEmitted).toBe(true);
+  });
+
+  test('should emit event width multiple handlers', () => {
+    const eventDispatcher = new EventDispatcher();
+
+    let eventEmitted1 = false;
+    let eventEmitted2 = false;
+    eventDispatcher.addEventListener('bar', () => {
+      eventEmitted1 = true;
+    });
+
+    eventDispatcher.addEventListener('bar', () => {
+      eventEmitted2 = true;
+    });
+
+    eventDispatcher.emit('bar');
+    expect(eventEmitted1).toBe(true);
+    expect(eventEmitted2).toBe(true);
+  });
+
+  test('should remove event listener by handler', () => {
+    const eventDispatcher = new EventDispatcher();
+
+    let eventEmitted = false;
+
+    const handler = () => {
+      eventEmitted = true;
+    };
+
+    eventDispatcher.addEventListener('bar', handler);
+    eventDispatcher.removeEventListener('bar', handler);
+    eventDispatcher.emit('bar');
+    expect(eventEmitted).toBe(false);
+  });
+
+  test('should remove event listener by name', () => {
+    const eventDispatcher = new EventDispatcher();
+
+    let eventEmitted = false;
+
+    eventDispatcher.addEventListener('bar', () => {
+      eventEmitted = true;
+    });
+
+    eventDispatcher.removeEventListener('bar');
+    eventDispatcher.emit('bar');
+    expect(eventEmitted).toBe(false);
```

Comment:
I think we can reference this style to test custom event dispatcher 
https://stackoverflow.com/questions/59957285/mock-a-custom-event-emitter-with-jest

Reply:
@mojosoeun 
Thank you for your comment. I modified it as you said
I am lacking a lot in this part. I ask for confirmation🙇